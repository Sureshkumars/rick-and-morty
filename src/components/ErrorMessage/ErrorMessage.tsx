import React from "react";
import { Text, VStack, Image, Box, Link, Button } from "@chakra-ui/core";
interface IProps {
  message: string;
  showHomeCTA: boolean;
}
export const ErrorMessage: React.FC<IProps> = (props: IProps) => {
  return (
    <VStack spacing={8} marginTop="10">
      <Box alignItems="Center" textAlign="center">
        <Image
          src="/images/no-results.gif"
          alt="Error Occurred"
          marginTop="5"
          marginBottom="5"
          maxW="280px"
          fallbackSrc="/images/character-fallback.jpeg"
        />
        <Text marginBottom="5" fontSize="xl">
          {props.message}
        </Text>
        {props.showHomeCTA && (
          <Link href="/" title="Return Home">
            <Button colorScheme="teal" variant="outline">
              Return Home
            </Button>
          </Link>
        )}
      </Box>
    </VStack>
  );
};
