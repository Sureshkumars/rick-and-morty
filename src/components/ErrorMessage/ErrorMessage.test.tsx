import React from "react";
import { ErrorMessage } from "./ErrorMessage";
import { render } from "@testing-library/react";

it("ErrorPage snapshot", () => {
  const error = {
    message: "No Results Found",
    showHomeCTA: true,
  };
  const { asFragment }: any = render(<ErrorMessage {...error} />);
  expect(asFragment()).toMatchSnapshot();
});
