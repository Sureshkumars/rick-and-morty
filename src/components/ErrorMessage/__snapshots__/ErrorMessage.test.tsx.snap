// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage snapshot 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack css-1cvi3v3"
  >
    <div
      class="css-14b1q7p"
    >
      <img
        alt="Error Occurred"
        class="chakra-image__placeholder css-6d0ud2"
        src="/images/character-fallback.jpeg"
      />
      <p
        class="chakra-text css-1yylkzx"
      >
        No Results Found
      </p>
      <a
        class="chakra-link css-0"
        href="/"
        title="Return Home"
      >
        <button
          class="chakra-button css-oqfhsh"
          type="button"
        >
          Return Home
        </button>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
